astropy/io/misc/tests/test_hdf5.py

```diff
@@ -823,3 +838,17 @@ def test_overwrite_serialized_meta():
         t2 = Table.read(out, path='data')
         assert all(t3 == t2)
         assert t3.info(out=None) == t2.info(out=None)
+
+
+@pytest.mark.skipif('not HAS_H5PY')
```

Comment:
This update should work for every Table file format, so you can just write to `ascii.basic` format which has no dependence.

Reply:
This test was to exercise HDF5 support in particular. I think my idea was that since HDF5 IO is outsourced to h5py whereas (most of?) the other table formats are handled directly within astropy, this is testing that the tilde-path expansion happens before things are passed to h5py and serves as a regression test in case the current code structure is ever changed. Otherwise this test is basically the same as the one in `test_table.py` and I can drop this one if you still think it's redundant.

Reply:
OK, good point.